spanner/client.go

```diff
@@ -263,13 +276,17 @@ func NewClientWithConfig(ctx context.Context, database string, config ClientConf
 // Combines the default options from the generated client, the default options
 // of the hand-written client and the user options to one list of options.
 // Precedence: userOpts > clientDefaultOpts > generatedDefaultOpts
-func allClientOpts(numChannels int, userOpts ...option.ClientOption) []option.ClientOption {
+func allClientOpts(numChannels int, compression string, userOpts ...option.ClientOption) []option.ClientOption {
 	generatedDefaultOpts := vkit.DefaultClientOptions()
 	clientDefaultOpts := []option.ClientOption{
 		option.WithGRPCConnectionPool(numChannels),
 		option.WithUserAgent(fmt.Sprintf("spanner-go/v%s", internal.Version)),
 		internaloption.EnableDirectPath(true),
 	}
+	if compression == "gzip" {
+		clientDefaultOpts = append(clientDefaultOpts, option.WithGRPCDialOption(grpc.WithDefaultCallOptions(
```

Comment:
Would it be possible to add a test with the mock Spanner server that verifies that we actually receive this header?

Reply:
Added the unit test for checking the request headers, also adding screenshot to verify all the grpc requests will have the call option for setting compression 
<img width="1654" alt="Screenshot 2023-05-19 at 11 55 13 AM" src="https://github.com/googleapis/google-cloud-go/assets/18293335/12e146e8-1b92-46cf-bd41-0e88d0c36ccf">

Reply:
Thanks for the extra info and adding the test. LGTM!